File: laboratory 1/python/sum.py
import os
import sys
from multiprocessing import Process

def do_sum(path):
    _sum = 0
    with open(path, 'rb',buffering=0) as f:
        byte = f.read(1)
        while byte:
            _sum += int.from_bytes(byte, byteorder='big', signed=False)
            byte = f.read(1)
    print(f"{path} : {_sum}")

if __name__ == "__main__":
    paths = sys.argv[1:]
    threads = []
    
    for path in paths:
    #many error could be raised error. we don't care
        thread = Process(target=do_sum, args=(path, ))
        thread.start()
        threads.append(thread)


    for thread in threads:
        thread.join()
    

File: laboratory 1/java/Sum.java
import java.io.FileInputStream;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.List;
import java.util.LinkedList;

public class Sum {

    public static int sum(FileInputStream fis) throws IOException {
        
	int byteRead;
        int sum = 0;
        
        while ((byteRead = fis.read()) != -1) {
        	sum += byteRead;
        }

        return sum;
    }

    public static long sum(String path) throws IOException {

        Path filePath = Paths.get(path);
        if (Files.isRegularFile(filePath)) {
       	    FileInputStream fis = new FileInputStream(filePath.toString());
            return sum(fis);
        } else {
            throw new RuntimeException("Non-regular file: " + path);
        }
    }

    public static void main(String[] args) throws Exception {

        if (args.length < 1) {
            System.err.println("Usage: java Sum filepath1 filepath2 filepathN");
            System.exit(1);
        }

	//many exceptions could be thrown here. we don't care
        List<Task> tasks = new LinkedList<Task>();
        List<Thread> threads = new LinkedList<Thread>();
        for (String path : args) {
            //long sum = sum(path);
            //System.out.println(path + " : " + sum);
            Task t = new Task(path);
            Thread currentThread = new Thread(t, "thread of " + path);
            currentThread.start();
            tasks.add(t);
            threads.add(currentThread);
        }

        // join for all
        for (Thread t : threads) {
            t.join();
        }

        // printing
        for (Task t : tasks) {
            System.out.println(t.getPath() + " : " + t.getSum());
        }
    }
}
